moto/dynamodb/models/table.py

```diff
@@ -560,6 +562,51 @@ def _validate_item_types(
                         "Start of structure or map found where not expected"
                     )
 
+    def _validate_number_type(self, value: Any) -> None:
+        if isinstance(value, int):
+            # TODO int and float values actually raise a ParamValidationError
+            raise InvalidConversion
+        if isinstance(value, float):
+            raise InvalidConversion
+        if not isinstance(value, str):
+            # TODO This is probably bad too, but keep current behavior
+            return
+
+        non_numeric_error = (
+            f"The parameter cannot be converted to a numeric value: {value}"
+        )
+
+        # Quick check - if the value can't be converted to a Python float, consider it invalid
+        try:
+            float_val = float(value)
+        except ValueError:
```

Comment:
There's a chance that this might cause some false positives, but I figure if it's not a valid python float, it's probably not a valid DynamoDB number either.